pyfar/signals/deterministic.py

```diff
@@ -256,6 +393,491 @@ def exponential_sweep_time(n_samples, frequency_range, n_fade_out=90,
+        `n_samples`, e.g., a stop margin of 100 samples means that the sweep
+        ends at sample ``n_samples-100``. This is required, because the
+        frequency domain sweep synthesis has post-ringing in the time domain.
+    fade_in : int, optional
+        Duration of a squared sine fade-in in samples. The fade starts at the
+        first sample of the sweep that is closer than 60 dB to the absolute
+        maximum of the sweep time signal. The default ``None`` does not apply
+        a fade-in.
+    fade_out : int, optional
+        Duration of a squared cosine fade-out in samples. The fade ends at the
+        last sample of the sweep that is closer than 60 dB to the absolute
+        maximum of the sweep time signal. The default ``None`` does not apply
+        a fade-out.
+    sampling_rate : int, optional
+        The sampling rate in Hz. The default is ``44100``.
+    group_delay : boolean, optional
+        Return the analytical group delay of the sweep. This can be used to
+        compute the times at which distortion products appear. The default is
+        ``False``.
+
+    Returns
+    -------
+    sweep : Signal
+        The sweep signal. The Signal is in the time domain, has a maximum
+        absolute amplitude of 1 and the ``none`` FFT normalization
+        (see :py:func:`~pyfar.dsp.fft.normalization`).
+    group_delay_sweep : FrequencyData
+        The analytical group delay of the sweep in seconds as a single sided
+        spectrum. Only returned if `group_delay` is ``True``.
+
+    References
+    ----------
+    .. [#] S. Müller, P. Massarani. 'Transfer Function Measurement with Sweeps.
+           Directors Cut Including Previously Unreleased Material and some
+           Corrections. J. Audio Eng. Soc. 2001, 49 (6), 443-471.
+
+    Examples
+    --------
+
+    .. plot::
+
+        >>> import pyfar as pf
+        >>> magnitude = pf.dsp.filter.low_shelve(
+        ...     pf.signals.impulse(2**16), 500, 20, 2)
+        >>> magnitude = pf.dsp.filter.butterworth(magnitude, 8, 50, 'highpass')
+        >>> sweep = pf.signals.magnitude_spectrum_weighted_sweep(
+        ...     2**16, magnitude, 5000, 1000)
+        >>> pf.plot.time_freq(sweep)
+    """
+
+    return _frequency_domain_sweep(
+        n_samples=n_samples,
+        sweep_type=magnitude_spectrum,
+        frequency_range=[0, sampling_rate / 2],
+        butterworth_order=0,
+        start_margin=start_margin,
+        stop_margin=stop_margin,
+        fade_in=fade_in,
+        fade_out=fade_out,
+        sampling_rate=sampling_rate,
+        group_delay=group_delay)
+
+
+def linear_perfect_sweep(n_samples, sampling_rate=44100, group_delay=False):
+    """
+    Generate a perfect linear sweep in the frequency domain.
+
+    The perfect sweep is generated according to [#]_ and is used for adaptive
+    system identification. It is orthogonal to delayed versions of itself and
+    can be played back in a loop.
```

Comment:
everything can be played back in a loop :)

Reply:
removed it - although infinitely long signals are hard to play in a loop